### ISSUE TITLE
The performance of reading small file

### ISSUE THREAD

**Issue by cheyang** [Alluxio/alluxio#11153]
**Alluxio Version:**
2.2.0
**Describe the bug**

1.生成小文件， 数量为40504， 总共6G, 文件平均大小为152K

```
docker run -v /mnt/essd-small-file:/data --env COCO=1 --env THREADS=64 -itd --name=generate-coco-data registry.cn-huhehaote.aliyuncs.com/tensorflow-samples/coco-perf /app/generate_file.sh
```

2.部署Alluxio 并且将数据完全缓存后运行并发读文件,目前8进程下，单独进程的吞吐只能达到205MiB

```
docker run -v /alluxio-fuse:/data -itd --name=read-imagenet-1x8-8-alluxio-small --env WORKERS=8 --env WORKERS_PER_NODE=8 --env THREADS=8  registry.cn-shanghai.aliyuncs.com/tensorflow-samples/coco-perf-mpi:py2
/data/* processing 40504 items with 8 threads uses 30.7677240372s, avg 1316.4444647/s, avg 215701996.514B/s, avg 205.709454073MiB/s
/data/* processing 40504 items with 8 threads uses 30.7632141113s, avg 1316.63745711/s, avg 215733618.697B/s, avg 205.739611337MiB/s
/data/* processing 40504 items with 8 threads uses 30.7675671577s, avg 1316.45117706/s, avg 215703096.347B/s, avg 205.710502956MiB/s
/data/* processing 40504 items with 8 threads uses 30.7671790123s, avg 1316.46778484/s, avg 215705817.564B/s, avg 205.713098111MiB/s
/data/* processing 40504 items with 8 threads uses 30.7651407719s, avg 1316.55500296/s, avg 215720108.424B/s, avg 205.726726937MiB/s
/data/* processing 40504 items with 8 threads uses 30.7685780525s, avg 1316.40792535/s, avg 215696009.47B/s, avg 205.703744383MiB/s
/data/* processing 40504 items with 8 threads uses 30.7682728767s, avg 1316.42098217/s, avg 215698148.856B/s, avg 205.70578466MiB/s
/data/* processing 40504 items with 8 threads uses 30.7824981213s, avg 1315.81263614/s, avg 215598470.171B/s, avg 205.610723659MiB/s
```

3.反复执行文件拷贝

```
du -sh /alluxio-fuse/ffb31500-e2cf-4c3f-ab6a-b14b201a7b7c
152K	/alluxio-fuse/ffb31500-e2cf-4c3f-ab6a-b14b201a7b7c
time cat /alluxio-fuse/ffb31500-e2cf-4c3f-ab6a-b14b201a7b7c > /dev/null

real	0m0.003s
user	0m0.000s
sys	0m0.001s
```

从日志中看随机会有对应的方法，耗时超过1ms，可以看一下是否有优化的空间

```
2020-03-11 05:02:38,724 WARN  AlluxioFuseFileSystem - release(path=/ffb31500-e2cf-4c3f-ab6a-b14b201a7b7c) returned 0 in 1 ms (>=1ms)
2020-03-11 05:02:48,590 WARN  AlluxioFuseFileSystem - open(path=/ffb31500-e2cf-4c3f-ab6a-b14b201a7b7c) returned 0 in 1 ms (>=1ms)
2020-03-11 05:02:48,591 WARN  AlluxioFuseFileSystem - release(path=/ffb31500-e2cf-4c3f-ab6a-b14b201a7b7c) returned 0 in 1 ms (>=1ms)
2020-03-11 05:02:50,083 WARN  AlluxioFuseFileSystem - read(path=/ffb31500-e2cf-4c3f-ab6a-b14b201a7b7c,buf=jnr.ffi.provider.jffi.DirectMemoryIO[address=0x7f64b0023fd0],size=65536,offset=0) returned 65536 in 1 ms (>=1ms)
2020-03-11 05:02:53,483 WARN  AlluxioFuseFileSystem - read(path=/ffb31500-e2cf-4c3f-ab6a-b14b201a7b7c,buf=jnr.ffi.provider.jffi.DirectMemoryIO[address=0x7f646c044be0],size=65536,offset=0) returned 65536 in 1 ms (>=1ms)
```

**To Reproduce**
Steps to reproduce the behavior (as minimally and precisely as possible)

**Expected behavior**

**Urgency**
Describe the impact and urgency of the bug.

**Additional context**

**Comment by Binyang2014** [comment]
Wonder the benchmark is using sequential access or random access. Just encounter such issue. Seems random access even worth.

**Comment by cheyang** [comment]
> Wonder the benchmark is using sequential access or random access. Just encounter such issue. Seems random access even worth.

sequential access. You can also post your test.

**Comment by DdMad** [comment]
I have got similar issue for small files and random access. I use fio to test the performance and below is the test result:

Test command: `fio -name=xxxx -filename=<-alluxio fuse mount path-> -rw=<read/randread> -bs=16K -size=10G -ioengine=libaio -numjobs=10 -per_job_logs=1`

Test result for seq access: throughput for each process is about 245MB/s
Test result for random access:  throughput for each process is about 25MB/s


I also tried larger block size,  `... -bs=1M ...` and the performance improved a lot:
Test result for seq access: throughput for each process is about 600MB/s
Test result for random access:  throughput for each process is about 500MB/s

To sum up, when deal with smaller files, both seq access and random access are slow, and random access is a lot slower than seq access; when deal with larger files, the performance is much better and random access is a bit slower than seq access.

I wonder whether this is normal for alluxio

**Comment by apc999** [comment]
We have been working on improving the performance of reading small files.
See the latest whitepaper based on 2.7 release:
https://www.alluxio.io/resources/whitepapers/alluxio-for-machine-learning-deep-learning-in-the-cloud/

A related investigation effort: https://github.com/Alluxio/alluxio/issues/14028
